src/util/util_ext.c

```diff
@@ -240,3 +240,133 @@ errno_t sss_filter_sanitize(TALLOC_CTX *mem_ctx,
 {
     return sss_filter_sanitize_ex(mem_ctx, input, sanitized, NULL);
 }
+
+enum sss_trim_dn_state {
+    SSS_TRIM_DN_STATE_READING_NAME,
+    SSS_TRIM_DN_STATE_READING_VALUE,
+};
+
+
+
+/* There is similar function ldap_dn_normalize in openldap.
+ * To avoid dependecies across project we have this own func.
+ * Also ldb can do this but doesn't handle all the cases
+ */
+static errno_t sss_trim_dn(TALLOC_CTX *mem_ctx,
+                           const char *input,
+                           char **trimmed)
+{
+    int i = 0;
+    int o = 0;
+    int s;
+    char *output;
+    enum sss_trim_dn_state state = SSS_TRIM_DN_STATE_READING_NAME;
+
+    *trimmed = NULL;
+
+    output = talloc_array(mem_ctx, char, strlen(input) + 1);
```

Comment:
From some reason sss_filter_sanitize_ex() uses bigger buffer for similar purpose. Nod sure if it is needed here but maybe it is worth checking:
160     /* Assume the worst-case. We'll resize it later, once */
161     output = talloc_array(mem_ctx, char, strlen(input) * 3 + 1);

Reply:
This is not needed. In Trim dn we make string shorter (or the same). In sanitize we expand chars to \XY (1 char to 3).